src/Temporalio/Client/TemporalClientOptions.cs

```diff
@@ -0,0 +1,32 @@
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Options for a <see cref="TemporalClient" />.
+    /// </summary>
+    public class TemporalClientOptions
+    {
+        /// <summary>
+        /// Gets or sets the client namespace. Default is "default".
+        /// </summary>
+        public string Namespace { get; set; } = "default";
+
+        // TODO(cretz): public Converters.DataConverter DataConverter { get; set; }
+
+        /// <summary>
+        /// Create default options.
+        /// </summary>
+        public TemporalClientOptions()
+        {
+            // TODO(cretz): DataConverter = Converters.DataConverter.Default;
+        }
+
+        /// <summary>
+        /// Create a shallow copy of these options.
+        /// </summary>
+        /// <returns>A shallow copy of these options.</returns>
+        public object Clone()
```

Comment:
We need shallow copying of all of our options classes. Options were:

1. Use records as options, but that doesn't fit https://learn.microsoft.com/en-us/dotnet/core/extensions/options
2. Create a copy constructor, but that is prone to hand-typed mistakes
3. Impl shallow `ICloneable` and expect caller to cast

3 seems best (there are other nuances, but this is the gist of it). I wish I could just have immutable classes/records and use `with`, but it doesn't fit with the .NET config/options preferences.